harness/tests/experiment/keras/test_tf_keras_trial.py

```diff
@@ -509,3 +515,89 @@ def make_workloads() -> workload.Stream:
     )
     with pytest.raises(ValueError, match="incompatible"):
         controller.run()
+
+
+@pytest.mark.tensorflow
+@pytest.mark.tf1_cpu
+def test_rng_restore(tmp_path: Path):
+    checkpoint_dir = str(tmp_path.joinpath("checkpoint"))
+    latest_checkpoint = None
+    steps_completed = 0
+
+    def make_checkpoint() -> workload.Stream:
+        trainer = utils.TrainAndValidate()
+
+        yield from trainer.send(steps=1, validation_freq=1, scheduling_unit=1)
+
+        interceptor = workload.WorkloadResponseInterceptor()
+        yield from interceptor.send(workload.checkpoint_workload())
+        nonlocal latest_checkpoint, steps_completed
+        latest_checkpoint = interceptor.metrics_result()["uuid"]
+        steps_completed = trainer.get_steps_completed()
+
+    def make_workloads_with_metrics(metrics_storage: List) -> workload.Stream:
+        trainer = utils.TrainAndValidate()
+
+        yield from trainer.send(steps=5, validation_freq=1, scheduling_unit=1)
+        _, validation_metrics = trainer.result()
+
+        metrics_storage += validation_metrics
+
+    config_base = utils.load_config(utils.fixtures_path("keras_no_op/const.yaml"))
+    hparams = config_base["hyperparameters"]
+
+    example_path = utils.fixtures_path("keras_no_op/model_def.py")
+    trial_class = utils.import_class_from_module("NoopKerasTrial", example_path)
+    trial_class._searcher_metric = "validation_error"
+
+    trial_B_metrics = []
+    trial_C_metrics = []
+
+    trial_A_controller = utils.make_trial_controller_from_trial_implementation(
+        trial_class=trial_class,
+        hparams=hparams,
+        workloads=make_checkpoint(),
+        checkpoint_dir=checkpoint_dir,
+        trial_seed=325,
+    )
+
+    trial_A_controller.run()
+
+    # reset random seed after checkpointing
+    other_random_seed = random.randint(0, 1 << 31)
+    while other_random_seed == 325:
+        other_random_seed = random.randint(0, 1 << 31)
+
+    trial_A_controller._set_random_seeds(other_random_seed)
```

Comment:
why not simplify all this to just one:
```python
trial_A_controller._set_random_seeds(0)
```
?

I only care that the RNG state  can't accidentally pick up from where it left off.

Reply:
Yeah that's better. Updated all tests for this.